Src/Couchbase.Linq/QueryGeneration/N1QlHelpers.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Couchbase.Linq.QueryGeneration
+{
+    /// <summary>
+    /// Helpers for N1QL query generation
+    /// </summary>
+    public static class N1QlHelpers
+    {
+
+        /// <summary>
+        ///     Escapes a N1QL identifier using tick (`) characters
+        /// </summary>
+        /// <param name="identifier">The identifier to format</param>
+        /// <returns>An escaped identifier</returns>
+        public static string EscapeIdentifier(string identifier)
+        {
+            if (identifier == null)
+            {
+                throw new ArgumentNullException("identifier");
+            }
+
+            if (identifier.IndexOf('`') >= 0)
+            {
+                // This should not occur, and is primarily in place to prevent N1QL injection attacks
+                // So it isn't performance critical to perform this replace in the StringBuilder
+
+                identifier = identifier.Replace("`", "``");
+            }
+
+            var sb = new StringBuilder(identifier.Length + 2);
```

Comment:
Good question, let me do some performance comparisons.

Brant

On Saturday, August 8, 2015, Jeffry Morris <notifications@github.com> wrote:

> In Src/Couchbase.Linq/QueryGeneration/N1QlHelpers.cs
> <https://github.com/couchbaselabs/Linq2Couchbase/pull/25#discussion_r36580995>
> :
>
> > +        public static string EscapeIdentifier(string identifier)
> > +        {
> > +            if (identifier == null)
> > +            {
> > +                throw new ArgumentNullException("identifier");
> > +            }
> > +
> > +            if (identifier.IndexOf('`') >= 0)
> > +            {
> > +                // This should not occur, and is primarily in place to prevent N1QL injection attacks
> > +                // So it isn't performance critical to perform this replace in the StringBuilder
> > +
> > +                identifier = identifier.Replace("`", "``");
> > +            }
> > +
> > +            var sb = new StringBuilder(identifier.Length + 2);
>
> Wouldn't a simple concat perform better than using StringBuilder?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/couchbaselabs/Linq2Couchbase/pull/25/files#r36580995>.
>


-- 
Thanks,
Brant Burnett
CenterEdge Software
(336) 598-5934
http://centeredgesoftware.com